CosmosToRedis/CosmosToRedis.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using StackExchange.Redis;
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.Design;
+using System.Drawing;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Threading.Tasks;
+using static System.Net.Mime.MediaTypeNames;
+using Container = Microsoft.Azure.Cosmos.Container;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Redis.Samples
+{
+    public record ListData
+    (
+        string id,
+        List<string> value
+    );
+
+    public static class CosmosToRedis
+    {
+        public const string localhostSetting = "redisLocalhost";
+        private static readonly IDatabase cache = ConnectionMultiplexer.Connect(Environment.GetEnvironmentVariable(localhostSetting)).GetDatabase();
+
+        public const string Endpoint = "Endpoint";
+
+        [FunctionName("CosmosToRedis")]
+        public static void Run([CosmosDBTrigger(
+        databaseName: "databaseName",
+        containerName: "containerName",
+        Connection = "Endpoint",
+        LeaseContainerName = "leases")]IReadOnlyList<ListData> input, ILogger log)
+        {
+            if (input == null || input.Count <= 0) return;
+
+            Console.WriteLine(input);
+
+            foreach (var value2 in input)
+            {
+                RedisValue[] redisValues = Array.ConvertAll(value2.value.ToArray(), item => (RedisValue)item);
```

Comment:
For my own understanding: if you push more than one item to the list will there be two items in the `input` list or two items n this `redisValues` list?

Reply:
Two in the input. The RedisValue[] is basically converting the IReadOnlyList<ListData> input into an array.

Reply:
You use `value2.value.ToArray` in this case `value2` is one of the `ListData` entries in the `IReadOnlyList input`. `value2.value` is the `List<string> value` attribute of the `ListData` object. If I add two values to `listTest` which list contains two entries? The `IReadOnlyList<ListData> input` or `RedisValue[] redisValues`? And if it is the first one, then when would `redisValues` contain more than one value?

Reply:
To clairfy, the input has all the entries. The redisValues is getting the value of one entry and then accesses that value to push into the cache. The redisValues never contains more than one value at once, it is one entry at a time.

Reply:
Thank you for the clarification. Is there ever a case where the `RedisValue` is null? 
I think it would make sense to remove the for loop at line 50 as it will never happen more than once.

Reply:
It would not be null unless the value was manually changed in CosmosDB to null, which then that null value would be sent to the cache as well. I tried removing the for loop and just doing each entry, but there was accessing issues with redisValues and the only solution I found was using the foreach loop. I'll look again and try again!

Reply:
I think we should understand what happens in that loop, if it only loops once, a null check and then just accessing the first item should be the solution!